crates/biome_service/src/workspace/server.rs

```diff
@@ -331,6 +331,17 @@ impl WorkspaceServer {
         } else if document_file_source.is_none() && !DocumentFileSource::can_parse(path.as_path()) {
             None
         } else {
+            // Before we try to parse the file, we might not even have anything to do with it.
+            // If none of the tools for this file are enabled, we can just skip it.
+            let settings = self.projects.get_settings(project_key);
+            let should_process = DocumentFileSource::formatter_enabled(&path, settings.as_ref())
+                || DocumentFileSource::linter_enabled(&path, settings.as_ref())
+                || DocumentFileSource::assist_enabled(&path, settings.as_ref());
```

Comment:
I'm not really sure this is the real fix. The scanner needs to analyze every file that can understand, even when it's ignored. 

This means that with this change, the scanner won't be able to scan `node_modules` because they are always ignored, and Biome needs to scan the node modules.

Reply:
That is really unfortunate. I'll close this in favor of a parser fix.